lib/matplotlib/testing/decorators.pyi

```diff
@@ -0,0 +1,25 @@
+from collections.abc import Callable, Sequence
+from pathlib import Path
+from typing import Any, TypeVar
+from typing_extensions import ParamSpec
+
+from matplotlib.figure import Figure
+from matplotlib.typing import RcStyleType
+
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+
+def remove_ticks_and_titles(figure: Figure) -> None: ...
+def image_comparison(
+    baseline_images: list[str] | None,
+    extensions: list[str] | None = ...,
+    tol: float = ...,
+    freetype_version: tuple[str, str] | str | None = ...,
+    remove_text: bool = ...,
+    savefig_kwarg: dict[str, Any] | None = ...,
+    style: RcStyleType = ...,
```

Comment:
Adjusting the stubtest allow list reveals that this type is not quite as permissive as we need it to be here. It allows `list[str]`, but the default parameter is (naturally) `tuple[str]`

https://github.com/matplotlib/matplotlib/blob/07afabad772fa3084d1575615016123de4c3152d/lib/matplotlib/typing.py#L58-L60

Should be adjusted to allow Sequence instead of just `list`


(As well as propegating the fact that `testing` is typed to `ci/mypy-stubtest-allowlist.txt`):

```diff
diff --git a/ci/mypy-stubtest-allowlist.txt b/ci/mypy-stubtest-allowlist.txt
index 58daf948c0..ac85209a99 100644
--- a/ci/mypy-stubtest-allowlist.txt
+++ b/ci/mypy-stubtest-allowlist.txt
@@ -6,7 +6,7 @@ matplotlib._.*
 matplotlib.rcsetup._listify_validator
 matplotlib.rcsetup._validate_linestyle
 matplotlib.ft2font.*
-matplotlib.testing.*
+matplotlib.testing.jpl_units.*
 matplotlib.sphinxext.*
 
 # set methods have heavy dynamic usage of **kwargs, with differences for subclasses
```